hw/dv/tools/dvsim/vcs.hjson

```diff
@@ -167,7 +167,12 @@
                      "-dir {cov_merge_db_dir}",
                      "-line nocasedef",
                      "-format both",
-                     "-elfile {vcs_cov_excl_files}",
+                     '''{eval_cmd} opts=`echo {vcs_cov_excl_files}`; \
+                     if [ "$opts" != "" ]; then \
+                       opts="-elfile $opts"; \
+                       echo $opts;
+                     fi
+                     ''',
```

Comment:
For next time, an easier way to get the same effect would be something like this:
```suggestion
                     '{eval_cmd} echo "{vcs_cov_excl_files}" | xargs -r echo -elfile',
```
The trick is to (ab)use xargs's `-r` argument:
```
rupert@halibut /s/l/opentitan (otbn)> echo 'a b c' | xargs -r echo -elfile
-elfile a b c
rupert@halibut /s/l/opentitan (otbn)> echo '' | xargs -r echo -elfile
```
Technically, this isn't bullet-proof (because `{vcs_cov_excl_files}` could be too long for a command line), but that is also an issue with the code here and I suspect other things will fail miserably if that happens!

Reply:
Incidentally, that trick will also only work if the exclusion files are devoid of special characters. Again, I suspect this isn't the only thing that would break in that case. But I really think some of this logic would be better in the Python: it feels a bit horrible doing arbitrary string processing by chaining together improperly escaped shell commands in a macro expansion meta-language.

Reply:
Thanks for sharing this magic trick. Will keep my original approach which is longer but more readable

Reply:
There is an example with `sed` as well on line 147 above.

Reply:
@sriyerg thanks. Let me open another PR to follow that example